samples/project-geometry/src/main/java/com/esri/arcgismaps/sample/projectgeometry/components/ProjectGeometryViewModel.kt

```diff
@@ -0,0 +1,116 @@
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.symbology.PictureMarkerSymbol
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.arcgismaps.mapping.view.SingleTapConfirmedEvent
+import com.arcgismaps.toolkit.geoviewcompose.MapViewProxy
+import com.esri.arcgismaps.sample.projectgeometry.R
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import java.util.Locale
+
+class ProjectGeometryViewModel(val app: Application) : AndroidViewModel(app) {
+    // create a map with a navigation night basemap style
+    val arcGISMap = ArcGISMap(BasemapStyle.ArcGISStreetsNight).apply {
+        // set the default viewpoint to Redlands,CA
+        initialViewpoint = Viewpoint(34.058, -117.195, 5e4)
+    }
+    // create a MapViewProxy to interact with the MapView
+    val mapViewProxy = MapViewProxy()
+
+    // Create a message dialog view model for handling error messages
+    val messageDialogVM = MessageDialogViewModel()
+
+    // setup the red pin marker image as bitmap drawable
+    private val markerDrawable: BitmapDrawable by lazy {
+        // load the bitmap from resources and create a drawable
+        val bitmap = BitmapFactory.decodeResource(app.resources, R.drawable.pin_symbol)
+        BitmapDrawable(app.resources, bitmap)
+    }
+
+    // setup the red pin marker as a Graphic
+    private val markerGraphic: Graphic by lazy {
+        // creates a symbol from the marker drawable
+        val markerSymbol = PictureMarkerSymbol.createWithImage(markerDrawable).apply {
+            // resize the symbol into a smaller size
+            width = 30f
+            height = 30f
+            // offset in +y axis so the marker spawned is right on the touch point
+            offsetY = 25f
+        }
+        // create the graphic from the symbol
+        Graphic(symbol = markerSymbol)
+    }
+
+    val graphicsOverlay = GraphicsOverlay().apply {
+        graphics.add(markerGraphic)
+    }
+
+    private val _infoText = MutableStateFlow(app.resources.getString(R.string.tap_to_begin))
+    val infoText: StateFlow<String> = _infoText
+
+    init {
+        viewModelScope.launch {
+            arcGISMap.load().onFailure { error ->
+                messageDialogVM.showMessageDialog(
+                    "Failed to load map",
+                    error.message.toString()
+                )
+            }
+        }
+    }
+
+    fun onMapViewTapped(event: SingleTapConfirmedEvent) {
+        event.mapPoint?.let { point ->
+            // update the marker location to where the user tapped on the map
+            markerGraphic.geometry = point
+            // set mapview to recenter to the tapped location
+            mapViewProxy.setViewpoint(Viewpoint(point))
```

Comment:
Use `setViewpointCenter` to get a nice animation to the new viewpoint.
```suggestion
            viewModelScope.launch {
                // set mapview to recenter to the tapped location
                mapViewProxy.setViewpointCenter(point)
            }
```

Reply:
This seems like a reasonable change, given without an animation it's very jarring